bazel/linux/defs.bzl

```diff
@@ -400,15 +400,17 @@ def _kernel_module_targets(*args, **kwargs):
 
         kwargs["name"] = name
         kwargs["kernel"] = kernel_tree_label
-        _gen_module_rule(arch, *args, **kwargs)
+        deps = kwargs["deps"] + [kdep.format(kernel) for kdep in kernel_specific_deps]
```

Comment:
instead of a format string here, could the rule surgically insert the kernel name in the correct location?  hiding the format string from the end user?

Reply:
yes, by doing something like `re.replace(":", ":{kernel}")` on the label names... a bit too "magic" for my taste, especially given the need to worry about canonicalization of labels (e.g. `//foo/bar:bar` will be minified to `//foo/bar` by tooling, which would mess with that process).

Your approach would definitely be easier to maintain / require the user to understand fewer details of how `kernel_module` names things, but that knowledge is already kind of required by the current design in order to have downstream dependencies not depend on the whole bundle (case in point: my other PR on this subject).

Reply:
Proposal:
1. Use named format, eg, `kdep.format(kernel_name=kernel)` in your for loop.
2. Don't introduce a new parameter, just use the normal `deps` parameter?

Reply:
Done in force-push.